### PR TITLE
Dinner Time!

### PR DESCRIPTION
**Pull request by Xackii**
## About The Pull Request

https://youtu.be/FqaZeh5uQqQ

30 minutes after shift start, dinner is announced at the station. Dinner lasts 5 minutes, during which crew members must come to a restaurant or coffeteria and eat something. 

At the end of 5 minutes, dinner ends and everyone receives a mood buff or debuff depending on what they spent dinner on. If they were not late for lunch (lateness is considered if you came to the restaurant and ate 2 minutes after the announcement of lunch) and at the end of lunch you were in the restaurant, you get a good mood buff. If you late on dinner you also gain good mood buff but but weaker. If you eat somthing outside dinner zones you gain mood debuff and if you haven't eaten anything you will get a strong mood debuff. if you came to restaruant, ate something and left before the end of lunch, then you will not receive the mood buff, but you will also not receive the debuff.
If you have an introvert quirk, then on the contrary you will receive a debuff if you ate in the cafeteria and a buff if you ate alone outside the cafeteria.

Dinner influence does not apply to those who are not at station z lvl, to the dead and to antagonists.

## Why It's Good For The Game

Allows for more fun interactions between station staff and each other. 
Gives more opportunity to roleplay as a hard worker who lives only for lunch and salary.
Expands the mechanics associated with mood and gives more gameplay to cooks and bartenders.
Station will have more reasons to be angry with the cooks for not preparing food.

## Changelog

:cl:
add: Dinner Time! Be ready cooks.
/:cl:

### CODE REVIEW

**Comment by YesterdaysPromise** [comment]
Introvert stocks rising.

**Comment by SimplySmiley123** [comment]
No.

**Comment by Dorsisdwarf** [comment]
Red Alert / shuttle call should probbably impact the need to be in the cafeteria

**Comment by CRITAWAKETS** [comment]
imagine if nukies are around and dinner time gets called

File: code/datums/mood_events/food_events.dm [hunk 1]
```diff
@@ -49,3 +49,38 @@
...
+	timeout = 10 MINUTES
+
+/datum/mood_event/out_dinner_room_eating
+	description = "I don't like eating outside the cafeteria, but work is more important."
+	mood_change = -4
+	timeout = 10 MINUTES
+
+/datum/mood_event/out_dinner_room_eating_introvert
+	description = "I like to have lunch alone."
+	mood_change = 8
```
Comment (CRITAWAKETS) [inline code/datums/mood_events/food_events.dm:80]: Why is this a higher mood bonus than the regular dinner bonus of 6?

**Comment by YuliceTheDruid** [comment]
Probably won't get in with the mood debuffs for not eating dinner, and the mood debuff for eating outside of the designated time will be hellaciously bad.

**Comment by DrDiasyl** [comment]
I'm not too fond of automated interactions.
It would be better if the Chef announced it himself, or at least got to choose dinner time with boundaries.

**Comment by Bibarra** [comment]
Let's not make chef a more miserable experience than it already is. This will just force chefs to either not sign up for the job or hyperoptimize their meal plan. Chefs who despise the system might even just not put anything out simply to spite the entire crew with a ten minute -8 mood debuff.

**Comment by Inari-Whitebear** [comment]
Why is it called lunch and dinner sometimes? Are those two diff events or are you just mixing up words?

**Comment by Sakamoto4ka** [comment]
> Let's not make chef a more miserable experience than it already is. This will just force chefs to either not sign up for the job or hyperoptimize their meal plan. Chefs who despise the system might even just not put anything out simply to spite the entire crew with a ten minute -8 mood debuff.

I think 5 mins enough for starving crew to make a burger from chefs

**Comment by DrDiasyl** [comment]
> this is just dumb. If people are in the middle of gimmicks or talking to each other or fighting we want them to continue doing that, not dropping everything to run to the bar to eat within 2 minutes of the announcement.

Comedy gold.

**Comment by SmArtKar** [comment]
Forcing human interactions by giving everyone a debuff that will make them go insane is really, really not a great idea.

**Comment by nianjiilical** [comment]
actually, i think this concept might work better as a random event

minimum 30 mins time to pop, only pops if theres a roundstart chef, can only pop once per round, and theres no or extremely little penalty for missing it but a nice mood buff for going

it'd basically be the friendly version of rad storm where it gives everyone an incentive to hang out in one spot for a few minutes

**Comment by necromanceranne** [comment]
> I'm not certain I like this concept _at all_ but I also think that 30 minutes is an extremely unrealistic time to set for this. That time limit to create enough food for the entire crew of sometimes 70+ people to all eat within a 5 minute window is going to put pressure on the chef to just make the quickest and most efficient food items possible as a gameplay requirement rather than spend their time... playing as a chef and being creative, although frankly I think this is kind of an unavoidable side effect of this feature existing with _any_ time limit.

It certainly puts the 'simulation' into the simulationist video game, but I'm not sure we should be exposing our players to the lunch time rush IRL chefs are subjected too on a daily basis. Unless you are hoping to make chefs turn homicidal. 

(IRL anecdote; I once had to dodge a burning towel being thrown across the kitchen because a chef leaned in too close to a griddle and and it lit up the towel around his neck. The lunch time rush makes people do some stupid as shit things)

**Comment by dalmationer** [comment]
In my experience working food service, the lunch rush is extremely gruelling. I don't think that that's something that necessarily is _bad_ to explore in video games, especially in the soft-critique of corporate & factory culture that Space Station 13 represents, but it should be done with intentionality. I feel that at present, this will add a lot of stress to anyone who plays chef and doesn't either make the easy meals, or come in with a plan and work hard on it. We do have players like this, who come up with a meal plan for breakfast, lunch and dinner, but it's contingent on botany being on the ball and not just making weapons or absent, and on minimal disruption by antagonists or roleplay.

I can forsee a situation with this PR where chef players feel good that they made a massive volume of food for the lunch rush and were able to serve everyone. I can also forsee a situation where a latejoin chef or someone newer to the game finds that everyone has just rushed into the kitchen and is demanding food and calling them a shit cook.

If you like the idea of a time-limited dinner time, why not just have it trigger while the shift is on night lighting mode and specifically be a mood boost, rather than a malus? That way you would presumably have regular, smaller rushes rather than one great big one.

**Comment by SimplySmiley123** [comment]
On one hand adds to simulationism of the game I guess, on the other, it being forced on you or you get a hefty mood debuff sucks major ass, alongside providing a pretty strong way to grief the whole station by making them insane for the chef from not providing any food and should not get implemented.

**Comment by Rhials** [comment]
> actually, i think this concept might work better as a random event

Seconded, the intent of this idea would work a lot better as an incidental occurrence rather than something static, that will inevitably have people structuring their routines around. 

I think the mood debuff is also a bit of a misstep, as not eating already carries its own hefty debuff. Adding a further one for skipping dinner just forces people to engage with the system which can be frustrating if you were doing literally anything else. You can still offer people the carrot without having to resort to the stick.

Don't get me wrong, I like this concept. The roots of a great idea are here, it just needs a bit of tuning to work with the greater flow of a round.

**Comment by SimplySmiley123** [comment]
_Also_ also
This will abso-fucking-lutely screw over non-humans.
Chefs already ignore food items that non-human species would eat, making those rely on scavenging for food, this PR would make all non-humans get guaranteed hefty mood debuff because no chef in a million years would try and make non-human food on a time crunch when they already give zero fucks without any time limit.

**Comment by dalmationer** [comment]
one suggestion that a pal of mine brought up was like, what if the chef could trigger the event when they felt they had enough food? by dinging a little triangle maybe.

**Comment by DaydreamIQ** [comment]
Having this exist every round would just suck for everyone. Especially on rounds where the chefs are new / nonexistent

**Comment by Exester509** [comment]
1. This affecting plasmamen, who don't need to eat and will burn alive if they do (with exceptions) is bad
2. This is a big screwup for round-long space explorers and other similar adventurers
3. Some sort of way for the chef to visibly announce breakfast/lunch/dinner when he has enough food would be preferable

**Comment by SomeoneYouProbablyKnow** [comment]
Mood is already a contentious system, people (for better or for worse) care so much about move speed that they powergame the hell out of mood, turning it from an incentive system into a near mandatory thing they have to keep track of. People who HAVE to eat or drink no longer roleplay with the barman or chef, it becomes a dry mechanical interaction and forcing them to do it more won't help.

It would probably work better as a less frequent occurrence, doing this every round will get super tedious and fast, people are already ruining their fun by doing like 30 minutes worth of powergaming chores. Something like a random event or even a station trait with opt in service jobs like waiters and line cooks could work.

>one suggestion that a pal of mine brought up was like, what if the chef could trigger the event when they felt they had enough food? by dinging a little triangle maybe.

Or that, that actually rocks

Alternatively don't limit it to a location, there are lots of cool moments to be had in sharing a snack in the middle of the SM chamber. And cramming more than 30 people into cafeteria is too hectic for most things. But that sort of feature creeps into something akin to a mood multiplier whenever you eat with someone else in my mind and that's a whole different thing.

**Comment by Paxilmaniac** [comment]
I'm ideasguying here but what'd be cooler is if instead of a set time, chefs get a comically sized dinner bell they can ring and if anyone eats at the kitchen a short time after they get the mood buff. This way people aren't punished for not eating at a certain place at a certain time, and the chef doesn't have to rush to make a pile of food right before 30 minutes lest they get a hungry crowd that breaks the door down.

**Comment by Loafin34** [comment]
itd be better if there was like a lunch break random event that gives people an announcement of their work being postponed for the union's lunch and their stomachs growling, making them hungry. what i find is an issue is most foods work too well and i spend an hour before i can eat again which sucks

**Comment by CCC23** [comment]
https://youtu.be/7tgVwOIKjJA?si=k5wLvvwSPMAlyqVr

**Comment by Twaticus** [comment]
sorry i actually LOVE this idea but its probably not being implemented in the best way

gonna ideasguy here
should not be mandatory for either the chef or the crew

have some sort of announcement system in the kitchen that chef can activate, which would send an automated station announcement like "lunchtime in 10 minutes", "lunchtime in 5 minutes", "lunch is served", something like that. maybe an option for a menu being announced too?

if you eat in the cafeteria after lunch has been served then you get a nice buff (maybe breakroom too? or it could be anywhere but as long as its something the chef made it would count?)

not eating during lunch shouldnt have a negative impact, just make it an opt-in thing. 

would be perfect reskinned for bartenders as a happy hour too

**Comment by SimplySmiley123** [comment]
> Mood is already a contentious system, people (for better or for worse) care so much about move speed that they powergame the hell out of mood, turning it from an incentive system into a near mandatory thing they have to keep track of. People who HAVE to eat or drink no longer roleplay with the barman or chef, it becomes a dry mechanical interaction and forcing them to do it more won't help.
> 
> It would probably work better as a less frequent occurrence, doing this every round will get super tedious and fast, people are already ruining their fun by doing like 30 minutes worth of powergaming chores. Something like a random event or even a station trait with opt in service jobs like waiters and line cooks could work.
> 
> > one suggestion that a pal of mine brought up was like, what if the chef could trigger the event when they felt they had enough food? by dinging a little triangle maybe.
> 
> Or that, that actually rocks
> 
> Alternatively don't limit it to a location, there are lots of cool moments to be had in sharing a snack in the middle of the SM chamber. And cramming more than 30 people into cafeteria is too hectic for most things. But that sort of feature creeps into something akin to a mood multiplier whenever you eat with someone else in my mind and that's a whole different thing.

It genuinely would be far better if it wasn't a forced mood affecting interaction every god damn round.
Especially considering that only really humans would ever get the benefit from it and everyone else would just get shafted.

**Comment by SimplySmiley123** [comment]
> The problem is that mood buffs are so prevalent that without the debuff i don't think anyone would care, mechanics wise. You still have the social aspect but, maybe that's enough idk.
> 
> > and the chef doesn't have to rush to make a pile of food right before 30 minutes lest they get a hungry crowd that breaks the door down.
> 
> As much as i don't like this dinner idea, gotta admit that is very funny

And with the debuff, it becomes a mandatory mechanic, not a roleplay interaction.
And again, non-humans will get pile drived by this.

**Comment by Dorsisdwarf** [comment]
> And again, non-humans will get pile drived by this.

They can just bring their own lunch to the canteen? Or eat the food that's there anyway? Like, do you think that currently nonhuman players just... don't eat all round? Like, making everyone come to the cafeteria at once at 30 minutes EVERY round is too much but you make it sound like they're gonna be the main course.

**Comment by SomeoneYouProbablyKnow** [comment]
mmmm lizard stew

**Comment by Loafin34** [comment]
make it not target plasmamen lol

**Comment by Wallemations** [comment]
Not sure how I feel about there being a debuff, but the idea is fun
It's kinda a similar thing to the breakfast system we have

**Comment by EricZilla** [comment]
I like the idea of chef hitting the once per round big lunch button that ONLY a trueblooded joined as chef crewmember can hit, that signals lunch will be active for the next 15 minutes, anyone who eats in that 15 gets a novel buff, anyone who doesnt or cant, maybe gets a neutral moodlet.

(((I also think it should effect noncrew because its just funny to envision nukies stopping to eat, or ashwalkers hearing the lunch siren wailing from the lavaland base)))

**Comment by Eronymun** [comment]
Remove the debuff and make it a buff only, then make it something chefs can choose to use wherever they want once per round.

If it's gotta have a stick then -1 mood is good enough, but it would be better without one. -8 is just too much.

**Comment by SomeoneYouProbablyKnow** [comment]
> Remove the debuff and make it a buff only, then make it something chefs can choose to use wherever they want once per round.
> 
> If it's gotta have a stick then -1 mood is good enough, but it would be better without one. -8 is just too much.

I don't think you can influence people with a mood boost. Firstly for the same reason everyone is against the debuff in the pr discussions, a small boost you can ignore but a small debuff psychologically feels very bad. Secondly, there are so many readily available mood boosts in the game, anyone who cares about mood even remotely is better off showering and catching a high five than spending time on dinner.
If you want to make a mechanical incentive with mood it gotta be a debuff or no one will care.
BUT, this feature has a lot of "social power" same way station announcements do. So there might not be a reason for a mechanical incentive if most players will care more about the announcement than the mood thing (80% of casual players don't even know mood is a thing*)
*completely made up statistic

But also yes -8 is literally insane. It could be -2 or somethin

**Comment by Xackii** [comment]
-8 if you didn't ate anything. It's don't check where you eat and what it's just check if you eat something for 5 minutes. -4 If you ate outside dinner room, not a strong debuff but good enough for dinner ignorance. 0 if you came to dinner room ate something and leave so you don't spend your time a lot and have no buff and no debuff

**Comment by SomeoneYouProbablyKnow** [comment]
-8 is not an incentive for people to engage with the feature it's forcing them to under gunpoint. And like it's been a while last i last got in the weeds of those numbers but isn't -8 literally on the level of losing a limb (limb might be -12 but that's close)

Unrelated to this argument but you gonna need a lot of checks for people to not accidentally to he hit by -8 mood. Stuff like being injured, unconcious, detained, high (because that's funny); red alert, no chef etc etc.

**Comment by Xackii** [comment]
Made from -8 to -4, -4 to -2 and introvert alone eating from 8 to 6

**Comment by Mrfilms** [comment]
Conceptually interesting, but having a hard timer would lead to a nightmare of chefs being swamped. It might be a better idea to, instead of having a set time that a dinner announcement will occur, give chefs a one-use 'dinner bell' item that does exactly what you propose here, but at the chef's discretion. I could see that benefitting my gameplay as a chef, for instance, as I often end up with meticulously laid out dinners, ready for the crew to eat, but people only arrive in drips and drabs.

**Comment by Xackii** [comment]
This PR not only for cooks to work harder. Spacemans can eat any food not only cookmade it's just check where food been eaten. It's just fun thing to split everyone on the station together or for antags to make traps or found itself targets or for clown to start slip anyone who come to dinner and get beaten or for secs to make block post and check anyone for loot and e.t.c.

**Comment by Ghommie** [comment]
Honestly, I prefer what #85369 was proposing, which is to let cooks CHOOSE when to announce a meal. Beside breakfast and moodlets, which is something of a get-go bonus of having a cook making food from the get-go, I'm a tad reticent on having a fixed time to serve a meal that carries no promise that someone has even cooked it, unlike a button which has to be pressed by someone with kitchen access on their own initiative.

**Comment by Xackii** [comment]
If the dinner time is chosen, it will kill the whole point of the PR. If there is just a button that announces that the cook has ready-made food and this announcement does not affect the gameplay in any way, then this is simply a useless mechanic that is unlikely to be used, it is as if the cook simply writes into the walkie-talkie that the food is ready. If this is a random event, it will be even worse and will also kill the whole point of the PR. It's also will kil point of the PR if it hasn't tied to a specific time. You are looking at only one of the mechanics with a mood debuff, although you are required to simply eat at lunch in 5 minutes so as not to get this debuff. The point of the PR is to open up more opportunities for the stations to interact with each other, as I wrote above, and not to grief people’s mood or make a useless button.

**Comment by Ghommie** [comment]
I'm not asking for this to be a random event for the very same reason. I see no problem with the button having little to no mechanics tied to it as long as the cook has control whether the announcement or event is triggered. About the debuff, my point stands: I'm not going to oblige players to find food 30 minutes into the round because of a hardcoded event. This is not a Simon says game, the hunger mechanics is there, and if a player mob doesn't seek food after a while, they'll get hungry, slow and sad. That's how it works, and furthermore it doesn't require any dissociated feature that doesn't care about hunger to begin with. This is not pearl clutching.

**Comment by Xackii** [comment]
The event does not affect people in other levels. I wrote this in PR, you didn't read. I can make dinner influence don't give debuff and give only buff but i don't wanna make it as a button or not tied to specific time.

**Comment by Ghommie** [comment]
Oh, it doesn't? That's great, and sorry about, my mind was somewhat elsewhere, looking at other PRs too.

**Pull request closed without merging**